pallets/ddc-verification/src/lib.rs

```diff
@@ -2430,6 +2553,54 @@ pub mod pallet {
 			}
 		}
 
+		pub(crate) fn get_consensus_for_sub_aggregates(
+			cluster_id: &ClusterId,
+			era_id: DdcEra,
+			activities: Vec<SubAggregates>,
+			min_nodes: u16,
+			threshold: Percent,
+		) -> Result<(Vec<SubAggregates>, Vec<SubAggregates>), Vec<OCWError>> {
+			let mut customer_buckets: BTreeMap<ActivityHash, Vec<SubAggregates>> = BTreeMap::new();
+
+			// Flatten and collect all customer activities
+			for activity in activities.iter() {
+				customer_buckets
+					.entry(activity.get_consensus_id::<T>())
+					.or_default()
+					.push(activity.clone());
+			}
+
+			let mut consensus_activities = Vec::new();
+			let mut not_consensus_activities = Vec::new();
+			let mut errors = Vec::new();
+			let min_threshold = threshold * min_nodes;
+
+			// Check if each customer/bucket appears in at least `min_nodes` nodes
+			for (id, activities) in customer_buckets {
+				if activities.len() < min_nodes.into() {
+					not_consensus_activities.extend(activities);
+				} else if let Some(activity) =
+					Self::reach_consensus(&activities, min_threshold.into())
+				{
+					consensus_activities.push(activity);
+				} else {
+					errors.push(OCWError::ActivityNotInConsensus {
+						cluster_id: (*cluster_id),
+						era_id,
+						id,
+					});
+				}
+			}
+
+			log::info!("🏠👍 Sub-aggregates, which are in consensus for cluster_id: {:?} for era_id: {:?}:::  {:?}", cluster_id, era_id, consensus_activities);
```

Comment:
where this logs go? to the container logs? wouldn't you have too much and too large messages on real envs?